record_processor.py

```diff
@@ -0,0 +1,65 @@
+import base64
+import csv
+import io
+import json
+
+from nypl_py_utils.classes.avro_client import AvroDecoder
+from nypl_py_utils.functions.log_helper import create_log
+
+
+class RecordProcessor:
+    def __init__(self, schema_url):
+        self.logger = create_log("record_processor")
+        self.avro_decoder = AvroDecoder(schema_url)
+        self.schema = self.avro_decoder.get_json_schema(schema_url)
+
+    def process_record(self, record, output_format):
+        """
+        Maps records to decoded Avro. Method returns data in
+        desired output format for Firehose (JSON or CSV string)
+        """
+        binary_record_data = base64.b64decode(record["data"])
+        decoded_record_data = self.avro_decoder.decode_record(
+            binary_record_data)
+
+        if decoded_record_data is None:
```

Comment:
I don't think this will ever happen -- if avro_decoder fails it'll throw an error not return None